### PR TITLE
MongoDB Connector Permissions [HZ-2328]

### PR DESCRIPTION
**Pull request by TomaszGaweda**
Adds permission handling in MongoDB connector

Fixes HZ-2328

EE PR: https://github.com/hazelcast/hazelcast-enterprise/pull/5988

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible

### CODE REVIEW

**Comment by TomaszGaweda** [comment]
run-lab-run

**Comment by TomaszGaweda** [comment]
run-lab-run

**Review by ldziedziul — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/security/permission/ConnectorPermission.java [hunk 1]
```diff
@@ -61,6 +62,21 @@ public static ConnectorPermission jdbc(@Nullable String connectionUrl, String ac
...
+     * @param connectionDescription connection string or data connection name
+     * @param databaseName database name this permission is about, null if permission is granted to all databases
+     * @param collectionName collection name this permission is about, null if permission is granted to all collections
+     * @param action action this permission is about
+     */
+    public static ConnectorPermission mongo(@Nullable String connectionDescription,
+                                            @Nullable String databaseName, @Nullable String collectionName,
+                                            String action) {
+        String db = databaseName == null ? "*" : databaseName;
+        String col = collectionName == null ? "*" : collectionName;
```
Comment (kwart) [inline hazelcast/src/main/java/com/hazelcast/security/permission/ConnectorPermission.java:75]: The wildcards here don't seem correct. This factory method is used for checked permissions and not the granted ones. 
If we allow `null` parameters, then I would either use the empty string or a hardcoded constant value.

Comment (TomaszGaweda) [reply hazelcast/src/main/java/com/hazelcast/security/permission/ConnectorPermission.java:75]: My understanding was that if user did not provide database name / collection name, then he must have permission for all dbs/collections. This is a case when user creates MongoClient by himself and we don't know if there is any specific db/col we want to protect.

Comment (kwart) [reply hazelcast/src/main/java/com/hazelcast/security/permission/ConnectorPermission.java:75]: It still feels wrong to me. What about using a hardcoded value that can't be a real value?
See https://www.mongodb.com/docs/manual/reference/limits/#naming-restrictions
e.g. `$ANY$`

File: hazelcast/src/main/java/com/hazelcast/security/permission/ConnectorPermission.java [hunk 2]
```diff
@@ -61,6 +62,21 @@ public static ConnectorPermission jdbc(@Nullable String connectionUrl, String ac
...
+        String dbCol = db + "/" + col;
+        return new ConnectorPermission(MONGO_PREFIX + connectionDescription + ":" + dbCol, action);
```
Comment (kwart) [inline hazelcast/src/main/java/com/hazelcast/security/permission/ConnectorPermission.java:77]: Do you want to allow `null` to appear in the String? The `connectionDescription` is `@Nullable`.

File: hazelcast/src/main/java/com/hazelcast/security/permission/ConnectorPermission.java (hunk 1 above)
Comment (TomaszGaweda) [reply hazelcast/src/main/java/com/hazelcast/security/permission/ConnectorPermission.java]: https://github.com/hazelcast/hazelcast/pull/24467/commits/68f6b23867a18b2f961db1f7d054865956f7fa5f

File: hazelcast/src/main/java/com/hazelcast/security/permission/ConnectorPermission.java (hunk 2 above)
Comment (TomaszGaweda) [reply hazelcast/src/main/java/com/hazelcast/security/permission/ConnectorPermission.java]: https://github.com/hazelcast/hazelcast/pull/24467/commits/f1c978f3a24f735b10713f0e35535ebb726b1c35

**Review by kwart — APPROVED** [review]

**Pull request merged**